src/S3/S3ClientTrait.php

```diff
@@ -213,15 +216,38 @@ public function determineBucketRegionAsync($bucketName)
         return $handler($command)
             ->then(static function (ResultInterface $result) {
                 return $result['@metadata']['headers']['x-amz-bucket-region'];
-            }, static function (AwsException $exception) {
-                $response = $exception->getResponse();
+            }, function (AwsException $e) {
+                $response = $e->getResponse();
                 if ($response === null) {
-                    throw $exception;
+                    throw $e;
+                }
+
+                if ($e->getAwsErrorCode() === 'AuthorizationHeaderMalformed') {
+                    $region = $this->determineBucketRegionFromExceptionBody(
+                        $response->getBody()
+                    );
+                    if (!empty($region)) {
+                        return $region;
+                    }
                 }
+
                 return $response->getHeaderLine('x-amz-bucket-region');
             });
     }
 
+    private function determineBucketRegionFromExceptionBody($responseBody)
+    {
+        try {
+            $element = $this->parseXml($responseBody);
+            if (!empty($element->Region)) {
+                return (string)$element->Region;
+            }
+        } catch (\Exception $e) {
+            return null;
```

Comment:
More of a nit, but I would expect this method to return `false` on error if it's swallowing exceptions.